src/engine/Resources/Sound/AudioInstance.ts

```diff
@@ -7,6 +7,12 @@ import { AudioContextFactory } from './AudioContext';
  * Internal class for producing of AudioInstances
  */
 /* istanbul ignore next */
+
+/**
+ * Since AudioInstance imlpementation should use WebAudio only,
+ * should the AudioInstanceFactory only return WebAudioInstance?
```

Comment:
To avoid breaking existing code, let's continue returning the `AudioInstance` type. In effect the only implementation will be `WebAudioInstance`. 

Bonus points, but not required, It may simplify things to "merge"  `WebAudioInstance` and `AudioInstance` together into `AudioInstance`